docs/dev/clusteroperator.md

```diff
@@ -85,7 +91,7 @@ If the controller reaches 4.0.1, the conditions might be:
 If an error blocks reaching 4.0.1, the conditions might be:
 
 * `Failing` is true with a detailed message `Unable to apply 4.0.1: could not update 0000_70_network_deployment.yaml because the resource type NetworkConfig has not been installed on the server.`
-* `Available` is false with message `Cluster is unable to reach 4.0.1`
```

Comment:
Why did you change my examples?  These are actual failures.  Add new ones.

Reply:
> Why did you change my examples? These are actual failures. Add new ones.

Because I think the examples are currently incorrect.  During an upgrade if you get stuck, the operator status is `Available==true` because there isn't an outage.  Having `Available=false` mean "the operand is unavailable" provides a clear dimension to create a alerts on.  The example in master doesn't have clear distinction between "page me!" and "this is stuck" and in this particular case required a lot of intelligence that doesn't seem tractable.

Reply:
You are available=false because the operand (the nested operators) are also not available during an upgrade that is failing halfway through.

Reply:
what nested operators?

for a use case like the registry operator, the registry may or may not be available when the registry operator is failing to upgrade, that's up to the registry operator to determine (and in most cases i'd expect us to remain available, at the previous version).